app/assets/javascripts/admin/views/broken-links-report.js

```diff
@@ -0,0 +1,76 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  function BrokenLinksReport (module) {
+    this.module = module
+  }
+
+  BrokenLinksReport.prototype.init = function () {
+    if (!fetch) return
+    var button = this.module.querySelector('button')
+    var authenticityTokenField = this.module.querySelector('[name="authenticity_token"]')
+    if (button && authenticityTokenField) this.setupEventListener(button, authenticityTokenField.value)
+    var refreshLink = this.module.querySelector('.js-broken-links-refresh')
+    if (refreshLink) {
+      this.setupPolling(refreshLink)
+    }
+  }
+
+  BrokenLinksReport.prototype.setupEventListener = function (button, authenticityToken) {
+    var action = this.module.querySelector('.js-broken-links-form').action
+    button.addEventListener('click', function (event) {
+      event.preventDefault()
+      fetch(action + '.json', {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({ authenticity_token: authenticityToken })
+      }).then(function (response) { return response.json() })
+        .catch(function () { location.reload() })
+        .then(function (json) {
+          this.replaceContentsAndCopyClasses(json.html)
+          this.init()
+        }.bind(this))
+    }.bind(this))
+  }
+
+  BrokenLinksReport.prototype.setupPolling = function (refreshLink) {
+    refreshLink.hidden = true
+
+    var retries = 10
+    var retry = function () {
+      retries -= 1
+      if (retries === 0) {
+        refreshLink.hidden = false
+        return
+      }
+
+      setTimeout(function () {
+        this.poll(refreshLink.href, retry)
+      }.bind(this), 2000)
+    }.bind(this)
+
+    retry()
+  }
+
+  BrokenLinksReport.prototype.poll = function (href, retry) {
+    fetch(href + '.json').then(function (response) { return response.json() })
+      .then(function (json) {
+        if (json.inProgress) {
+          retry()
+        } else {
+          this.replaceContentsAndCopyClasses(json.html)
+          this.init()
+        }
+      }.bind(this))
+      .catch(function () { retry() })
+  }
+
+  BrokenLinksReport.prototype.replaceContentsAndCopyClasses = function (html) {
+    this.module.innerHTML = html
+    this.module.classList = this.module.firstChild.classList
+    this.module.firstChild.outerHTML = this.module.firstChild.innerHTML
```

Comment:
Is there a reason why we need to do this? Can we not just have a message placeholder div which we place?

Reply:
This way the rendering is the same for in progress and the final report.

We need it in the template anyway since that will be displayed if the page is loaded while a report is being generated or javascript is not being used.

Reply:
Is there a card for this?